File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;
        int maxProfit = 0;

        for(int i = 0; i < prices.length; i++){
            minPrice = Math.min(prices[i],minPrice);
            int currProfit = prices[i] - minPrice;
            maxProfit = Math.max(currProfit,maxProfit);
        }
    return maxProfit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = float('inf')
        maxProfit = 0

        for i in range(len(prices)):
            minPrice = min(prices[i],minPrice)
            currProfit = prices[i] - minPrice
            maxProfit = max(currProfit,maxProfit)
        return maxProfit